### PR TITLE
feat: `@vueuse/head` v1

### PR DESCRIPTION
**Pull request by harlan-zw**
### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation or readme)
- [x] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [x] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

Read the v1 @vueuse/head post for context https://harlanzw.com/blog/vue-use-head-v1

### Nuxt & @vueuse/head

I have pushed this PR up to try and get it in before the v3 stable.

The DOM patching logic is more complex than the previous implementation and it's quite possible there are issues that won't be uncovered until wide testing. So while my personal testing has gone quite well, I'll leave this as a draft PR until I have a playground ready and can do some more testing.

### Next Steps

In terms of development, once this is merged then I can look at implementing the rest of the new features unlocked by this:
- add shortcut composables (includes useTagMetaFlat, solves the Meta SEO of this [discussion](https://github.com/nuxt/framework/discussions/1823))
- add `useServerHead`, opens the door for 0kb runtime head
- expose hooks, this will be powerful for module authors

## Migration Guide

(copied from the release notes)

⚠️ Breaking changes are minimal, but there are some changes to be aware of.

Please report any issues you find and they will fixed be promptly.

You may consider using [@unhead/vue](https://unhead.harlanzw.com/integrations/vue/setup) directly if you [don't need @vueuse/head](https://unhead.harlanzw.com/integrations/vue/vueuse-head).

### Verify your tags

The new DOM patching algorithm has not been tested in all possible scenarios, it's possible that there are unforeseen edge cases.  

### `htmlAttrs` and `bodyAttrs` merge strategy

If you had built your code around with the assumption that setting `htmlAttrs` or `bodyAttrs` would clear the old tags, this is now different.

```ts
// old
useHead({
  htmlAttrs: {
    class: 'my-class',
  },
})

useHead({
  htmlAttrs: {
    class: 'new-class',
  },
})

// <html class="new-class">
```

```ts
// new
useHead({
  htmlAttrs: {
    class: 'my-class',
  },
})

useHead({
  htmlAttrs: {
    class: 'new-class',
  },
})

// <html class="my-class new-class">
```

Check the [documentation](https://unhead.harlanzw.com/guide/guides/handling-duplicates#tagduplicatestrategy) to learn more.

### Duplicate tags in the same entry allowed

To make duplicate tag handling more intuitive, duplicate tags are now allowed in the same entry.

Having these metas across `useHead`'s will still dedupe as before.

```ts
// old
useHead({
  meta: [
    { name: 'description', content: 'foo' },
    { name: 'description', content: 'bar' },
  ],
})

// only the last tag is rendered
// <meta name="description" content="bar">
```

```ts
// old
useHead({
  meta: [
    { name: 'description', content: 'foo' },
    { name: 'description', content: 'bar' },
  ],
})

// both are rendered
// <meta name="description" content="foo">
// <meta name="description" content="bar">
```

### 📝 Checklist

- [ ] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: packages/nuxt/src/head/module.ts [hunk 1]
```diff
@@ -29,9 +29,6 @@ export default defineNuxtModule({
       })
     }
 
-    // Add mixin plugin
-    addPlugin({ src: resolve(runtimeDir, 'mixin-plugin') })
```
Comment (harlan-zw) [inline packages/nuxt/src/head/module.ts:33]: unhead ships with native options API support

https://github.com/harlan-zw/unhead/blob/main/packages/vue/src/createHead.ts#L44

File: packages/schema/src/types/meta.ts [hunk 2]
```diff
@@ -29,70 +29,3 @@ export type AppHeadMetaObject = MetaObjectRaw & {
    */
   viewport?: string
 }
-
-export interface MetaObject {
```
Comment (harlan-zw) [inline packages/schema/src/types/meta.ts:33]: I couldn't figure out why this needed to be exposed by @nuxt/schema exactly. The reactive object is only passed used as an argument for the `useHead` function, `nuxt.config.ts` uses the non-reactive schema

**Comment by harlan-zw** [comment]
I figured that having a proper v1 of @vueuse/head would help move this PR along, so it's now released.

The first draft of documentation for https://unhead.harlanzw.com/ is complete.

What's remaining is for me to spin up a playground and do some final nuxt-specific testing with edge cases, I hope to have that done by this time tomorrow. Aim to have it in a RC if the code is good

**Comment by harlan-zw** [comment]
Looks like I goofed up with the v1 release, didn't realise the constraint would jump from rc to 1.0.0. So in fact the nuxt core is now using v1 @vueuse/head for any new installs.

I updated it to be fully backwards compatible to unblock any users, so this PR will now just be switching over from the deprecated functions

**Comment by manniL** [comment]
@harlan-zw Will you release a v2 shortly then removing backwards compat?

**Comment by harlan-zw** [comment]
> @harlan-zw Will you release a v2 shortly then removing backwards compat?

Nope, seems a backwards compat build for v3 stable is ideal, there are some ecosystem dependencies that also rely on the old API. (vuetify)

**Pull request closed without merging**